### DISCUSSION TITLE
Welcome to go-ios Discussions!

### DISCUSSION THREAD

**Discussion by danielpaulus** [danielpaulus/go-ios#83] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by jobeylev** [comment]
Hi, I'm Jonathan, iOS tech lead responsible for building a custom iOS automation solution for a very major product. Your work has helped us so much!

**Reply by danielpaulus** [reply]
Wow that is great to hear! 
If you have feature requests, ideas or suggestions feel free to share them :-) 
I would like to keep improving the project so it can be a great platform for anyone trying to build anything iOS device automation.
Have you seen the new REST API? 
If there is an interest, I can share insights on how to build a relatively stable device farm with this stuff. 

Also, if your company has an open source budget, a small sponsorship would be cool ;-)

**Reply by jobeylev** [reply]
Thanks Daniel, we have a very bespoke environment and your command line option integrates with it very nicely. I’ve not considered the implications of a test API yet. 

I will pass along your offer re the device farm — I’m not sure what our needs are in that department. 

I will be sure to check whether we have an open source budget. No project deserves it more from our team!

**Comment by ghost** [comment]
Hey there!

**Comment by adwaittathe** [comment]
Hey, how can I get the screen height or width information for non core device (OS <= 16)

I am using the command below to get the screen info.
`ios info display --udid <UDID>`

Its giving me this error
`{"err":"NewDeviceInfo: failed to connect to XPC service: ConnectToXpcServiceTunnelIface: Cannot connect to com.apple.coredevice.deviceinfo, missing tunnel address and RSD port. To start the tunnel, run `ios tunnel start`","level":"fatal","msg":"Can't connect to deviceinfo service","time":"2024-11-25T23:51:00-08:00"}`

I tried starting the ios tunnel using the command `ios tunnel start  --udid <UDID>` but its showing me this error 
`{"error":"manualPairingTunnelStart: unsupported iOS version 16.7.1","level":"warning","msg":"failed to start tunnel","time":"2024-11-25T22:58:14-08:00","udid":"19e2822676bd93a557917aff5ce617c3a52488d6"}`. 

From error it looks like its using core device commands for getting the screen display information. 

I am able to start the tunnel for core device and get the screen display information. Is there any other command to get this information for non core devices ?
